pyscriptjs/src/python/pyscript.py

```diff
@@ -497,4 +496,158 @@ def wrapper(class_):
+        self._show_warning(self.__message)
+        self.__warning_already_shown = True
+
+    def __getattr__(self, attr):
+        self._show_warning_maybe()
+        return getattr(self.__obj, attr)
+
+    def __call__(self, *args, **kwargs):
+        self._show_warning_maybe()
+        return self.__obj(*args, **kwargs)
+
+    def __iter__(self):
+        self._show_warning_maybe()
+        return iter(self.__obj)
+
+    def __getitem__(self, key):
+        self._show_warning_maybe()
+        return self.__obj[key]
+
+    def __setitem__(self, key, value):
+        self._show_warning_maybe()
+        self.__obj[key] = value
+
+
+class PyScript:
+    """
+    This class is deprecated since 2022.12.1.
+
+    All its old functionalities are available as module-level functions. This
+    class should be killed eventually.
+    """
+
+    loop = loop
+
+    @staticmethod
+    def run_until_complete(f):
+        run_until_complete(f)
+
+    @staticmethod
+    def write(element_id, value, append=False, exec_id=0):
+        write(element_id, value, append, exec_id)
+
+
+def _install_deprecated_globals_2022_12_1(ns):
+    """
+    Install into the given namespace all the globals which have been
+    deprecated since the 2022.12.1 release. Eventually they should be killed.
+    """
+
+    def deprecate(name, obj, instead):
+        message = f"Direct usage of <code>{name}</code> is deprecated. " + instead
+        ns[name] = DeprecatedGlobal(name, obj, message)
+
+    # function/classes defined in pyscript.py ===> pyscript.XXX
+    pyscript_names = [
+        "PyItemTemplate",
+        "PyListTemplate",
+        "PyWidgetTheme",
+        "add_classes",
+        "create",
+        "loop",
+    ]
+    for name in pyscript_names:
+        deprecate(
+            name, globals()[name], f"Please use <code>pyscript.{name}</code> instead."
+        )
+
+    # stdlib modules ===> import XXX
+    stdlib_names = [
+        "asyncio",
+        "base64",
+        "io",
+        "sys",
+        "time",
+        "datetime",
+        "pyodide",
+        "micropip",
+    ]
+    for name in stdlib_names:
+        obj = __import__(name)
+        deprecate(name, obj, f"Please use <code>import {name}</code> instead.")
+
+    # special case
+    deprecate(
+        "dedent", dedent, "Please use <code>from textwrap import dedent</code> instead."
+    )
+
+    # these are names that used to leak in the globals but they are just
+    # implementation details. People should not use them.
+    private_names = [
+        "eval_formatter",
+        "format_mime",
+        "identity",
+        "render_image",
+        "MIME_RENDERERS",
+        "MIME_METHODS",
+    ]
+    for name in private_names:
+        obj = globals()[name]
+        message = (
+            f"<code>{name}</code> is deprecated. "
+            "This is a private implementation detail of pyscript. "
+            "You should not use it."
+        )
+        ns[name] = DeprecatedGlobal(name, obj, message)
```

Comment:
why not use the `deprecate` method here? just like others.
I mean, why use `ns[name] = DeprecatedGlobal(name, obj, message)` explicitly when that functionality is already abstracted under `deprecate` method.

Reply:
because I wanted to use a completely different message

Reply:
Yeah, but then, that message can simply be passed as one of the parameters to the deprecate function.

Reply:
As far as I recall, the deprecate function already expects the message as its argument. So even the "different" message can be defined separately and then passed to it. Not sure I understand how a different message demands not using it 😅

Reply:
I think I got what you meant, this is because the message is about `private implementation` and not about `deprecation`. But the other one is about deprecation I guess.

Reply:
Resolving this, but leaving the other one open.